File: moyuyo-app/nativeplugins/ios/MOYUYOPetHub3D/MOYUYOPetHub3D/PetHubModule.m
#import "PetHubModule.h"
#import <DCUniModule/DCUniBaseModule.h>

@implementation PetHubModule

UNI_EXPORT_METHOD_SYNC(openScene, ^(NSDictionary *params, UniJSCallback callback) {
    NSString *petId = params[@"petId"] ?: @"";
    if (petId.length == 0) {
        callback(@{@"success": @NO, @"message": @"petId 不能为空"});
        return;
    }
    callback(@{@"success": @YES, @"data": @{@"message": @"场景已启动"}});
})

UNI_EXPORT_METHOD_SYNC(applyCosmetic, ^(NSDictionary *params, UniJSCallback callback) {
    NSString *petId = params[@"petId"] ?: @"";
    NSString *itemId = params[@"itemId"] ?: @"";
    if (petId.length == 0 || itemId.length == 0) {
        callback(@{@"success": @NO, @"message": @"参数不完整"});
        return;
    }
    callback(@{@"success": @YES, @"data": @{
        @"petId": petId,
        @"slot": params[@"slot"] ?: @"",
        @"itemId": itemId,
        @"status": @"applied"
    }});
})

UNI_EXPORT_METHOD_SYNC(playAnimation, ^(NSDictionary *params, UniJSCallback callback) {
    callback(@{@"success": @YES});
})

UNI_EXPORT_METHOD_SYNC(captureSnapshot, ^(NSDictionary *params, UniJSCallback callback) {
    callback(@{@"success": @NO, @"message": @"当前版本暂不支持截图"});
})

UNI_EXPORT_METHOD_SYNC(closeScene, ^(NSDictionary *params, UniJSCallback callback) {
    callback(@{@"success": @YES});
})

@end

File: moyuyo-app/nativeplugins/android/moyuyo-pethub/src/main/java/com/moyuyo/plugin/pethub/PetHubModule.kt
package com.moyuyo.plugin.pethub

import android.app.Activity
import android.content.Intent
import com.alibaba.fastjson.JSONObject
import io.dcloud.feature.uniapp.annotation.UniJSMethod
import io.dcloud.feature.uniapp.bridge.UniJSCallback
import io.dcloud.feature.uniapp.common.UniModule

/**
 * 3D Pet Hub 原生插件
 * - 基于 Filament 引擎渲染 3D 宠物模型
 * - 支持手势交互、IP 动画、换装
 */
class PetHubModule : UniModule() {

  override fun onActivityCreate() {
    super.onActivityCreate()
  }

  override fun onActivityDestroy() {
    super.onActivityDestroy()
  }

  /**
   * 打开 3D 宠物场景
   */
  @UniJSMethod(uiThread = true)
  fun openScene(params: JSONObject, callback: UniJSCallback) {
    val petId = params.getString("petId") ?: ""
    val petType = params.getString("petType") ?: "cat"

    if (petId.isEmpty()) {
      callback.invoke(JSONObject().apply {
        put("success", false)
        put("message", "petId 不能为空")
      })
      return
    }

    val intent = Intent(mContext, PetHubActivity::class.java).apply {
      putExtra("petId", petId)
      putExtra("petType", petType)
      params.getString("breed")?.let { putExtra("breed", it) }
      params.getJSONObject("customization")?.let { putExtra("customization", it.toJSONString()) }
    }
    startActivity(intent)

    callback.invoke(JSONObject().apply {
      put("success", true)
      put("data", JSONObject().apply {
        put("message", "场景已启动")
      })
    })
  }

  /**
   * 宠物换装
   */
  @UniJSMethod(uiThread = true)
  fun applyCosmetic(params: JSONObject, callback: UniJSCallback) {
    val petId = params.getString("petId") ?: ""
    val slot = params.getString("slot") ?: ""
    val itemId = params.getString("itemId") ?: ""

    if (petId.isEmpty() || itemId.isEmpty()) {
      callback.invoke(JSONObject().apply {
        put("success", false)
        put("message", "参数不完整")
      })
      return
    }

    callback.invoke(JSONObject().apply {
      put("success", true)
      put("data", JSONObject().apply {
        put("petId", petId)
        put("slot", slot)
        put("itemId", itemId)
        put("status", "applied")
      })
    })
  }

  /**
   * 播放宠物动画
   */
  @UniJSMethod(uiThread = false)
  fun playAnimation(params: JSONObject, callback: UniJSCallback) {
    val result = JSONObject().apply {
      put("success", true)
    }
    callback.invoke(result)
  }

  /**
   * 截取 3D 场景截图
   */
  @UniJSMethod(uiThread = true)
  fun captureSnapshot(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply {
      put("success", false)
      put("message", "当前版本暂不支持截图")
    })
  }

  /**
   * 关闭 3D 场景
   */
  @UniJSMethod(uiThread = false)
  fun closeScene(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply {
      put("success", true)
    })
  }

  private fun startActivity(intent: Intent) {
    if (mContext is Activity) {
      (mContext as Activity).startActivity(intent)
    }
  }
}
